src/transformers/tokenization_utils_base.py

```diff
@@ -776,7 +776,14 @@ def to(self, device: Union[str, "torch.device"]) -> "BatchEncoding":
             :class:`~transformers.BatchEncoding`: The same instance of :class:`~transformers.BatchEncoding` after
             modification.
         """
-        self.data = {k: v.to(device) for k, v in self.data.items()}
+
+        # This check catches things like APEX blindly calling "to" on all inputs to a module
+        # Otherwise it passes the casts down and casts the LongTensor containing the token idxs
+        # into a HalfTensor
+        if isinstance(device, str) or isinstance(device, torch.device):
+            self.data = {k: v.to(device=device) for k, v in self.data.items()}
+        else:
+            logger.warning("Attempting to cast a BatchEncoding to another type, %s", str(device))
```

Comment:
```suggestion
            logger.warning(f"Attempting to cast a BatchEncoding to another type, {str(device)}. This is not supported.")
```

Reply:
Sure, fine by me. I think I copied a logger statement from further down the file, so there are some others outside of this commit that need fixing in that file.